iroha_data_model/src/lib.rs

```diff
@@ -1126,10 +1143,30 @@ pub mod transaction {
             Hash::new(&bytes)
         }
 
+        /// Checks if number of instructions in payload exceeds maximum
+        pub fn check_instruction_len(&self, max_instruction_number: usize) -> Result<(), String> {
+            if self
+                .payload
+                .instructions
+                .iter()
+                .map(Instruction::len)
+                .sum::<usize>()
+                > max_instruction_number
+            {
+                return Err("Too many instructions in payload".to_owned());
+            }
+            Ok(())
+        }
+
         /// Sign transaction with the provided key pair.
         ///
         /// Returns `Ok(Transaction)` if succeeded and `Err(String)` if failed.
-        pub fn sign(self, key_pair: &KeyPair) -> Result<Transaction, String> {
+        pub fn sign(
+            self,
+            key_pair: &KeyPair,
+            max_instruction_number: usize,
+        ) -> Result<Transaction, String> {
+            self.check_instruction_len(max_instruction_number)?;
```

Comment:
I think there should not be any check on sign. As it is unrelated operation. If you want to add the check to the client library, you can simply check it on submit.

Reply:
Ok, will remove this check here